src/Internal/Wallet/Cip30Mock.purs

```diff
@@ -109,20 +133,31 @@ withCip30Mock (KeyWallet keyWallet) mock contract = do
     MockLode -> "LodeWallet"
     MockNuFi -> "nufi"
 
+type CollateralParams = { amount :: String }
+
 type Cip30Mock =
   { getNetworkId :: Effect (Promise Int)
-  , getUtxos :: Effect (Promise (Array String))
-  , getCollateral :: Effect (Promise (Array String))
+  , getUtxos ::
+      Fn2 (UndefinedOr String) (UndefinedOr Paginate)
+        (Promise (Nullable.Nullable (Array String)))
```

Comment:
Please remove Nullable dependency. untagged-union provides a `HasRuntimeType Null` instance, which makes it possible to use `Null |+| a` (very similar to  `UndefinedOr` from `untagged-union` that we already use).